src/3rdparty/CMakeLists.txt

```diff
@@ -7,6 +7,14 @@ ADD_SUBDIRECTORY(hiir)
 ADD_SUBDIRECTORY(rpmalloc)
 ADD_SUBDIRECTORY(weakjack)
 
+INCLUDE(CheckCXXPreprocessor)
+CHECK_CXX_DEFINE(IS_MINGW "__MINGW32__")
+IF(IS_MINGW)
```

Comment:
We seem to have three different ways to check for MinGW in use in the codebase. There's:
* This one, checking whether `__MINGW32__` is defined;
* CMake's built-in `MINGW` variable; and
* Checking whether `MINGW_PREFIX` is set, which we set ourselves in custom toolchain files.

I would guess, but haven't checked, that the `__MINGW32__` and `MINGW_PREFIX` approaches are older ones that were used before we were guaranteed to have CMake 3.2, which is when the `MINGW` variable was introduced. I would rather we used the built-in CMake variable, unless there's a particular reason not to.

Reply:
> CMake's built-in MINGW variable; and

I do remember going with this approach, but I switched to ``CHECK_CXX_DEFINE`` for some reason awhile back, (probably because the built-in ``MINGW`` variable wasn't supported with our CMake versions yet, not sure why I would have any other preference otherwise since it's the cleanest solution) but I'll try to use it again.